src/internal/repository-metadata-files/provider/create.go

```diff
@@ -1,22 +1,21 @@
 package provider
 
 import (
-	"fmt"
+	"path/filepath"
 	"registry-stable/internal/files"
 	"registry-stable/internal/provider"
 )
 
-func CreateMetadataFile(p provider.Provider) error {
-	repositoryFileData, err := BuildMetadataFile(p)
+func CreateMetadataFile(p provider.Provider, providerDataDir string) error {
+	repositoryFileData, err := BuildMetadataFile(p, providerDataDir)
 	if err != nil {
 		return err
 	}
 
-	filePath := getFilePath(p)
+	filePath := getFilePath(p, providerDataDir)
 	return files.WriteToFile(filePath, repositoryFileData)
 }
 
-func getFilePath(p provider.Provider) string {
-	shard := p.Namespace[0]
-	return fmt.Sprintf("providers/%c/%s/%s.json", shard, p.Namespace, p.ProviderName)
+func getFilePath(p provider.Provider, providerDataDir string) string {
+	return filepath.Join(providerDataDir, p.Namespace[0:1], p.Namespace, p.ProviderName+".json")
```

Comment:
Should the shard name be case-insensitive? I think so.

Basically, I'd use strings.ToLower for the shard.

Reply:
Done in e8f075dab560789e332c10e33a63c697867bcb74

We'd need to later refactor how we get the module/provider directory path